src/index.ts

```diff
@@ -0,0 +1,70 @@
+import { Duplex } from 'stream';
+
+export default class PortDuplexStream extends Duplex {
+  private _port: Record<string, any>;
+
+  constructor(port: Record<string, any>) {
+    super({ objectMode: true });
+    this._port = port;
+    this._port.onMessage.addListener((msg: any) => this._onMessage(msg));
+    this._port.onDisconnect.addListener(() => this._onDisconnect());
+  }
+
+  /**
+   * Callback triggered when a message is received from
+   * the remote Port associated with this Stream.
+   *
+   * @private
+   * @param msg - Payload from the onMessage listener of Port
+   */
+  _onMessage(msg: any): void {
+    if (Buffer.isBuffer(msg)) {
+      // delete msg._isBuffer;
+      const data = Buffer.from(msg);
+      this.push(data);
+    } else {
+      this.push(msg);
+    }
+  }
+
+  /**
+   * Callback triggered when the remote Port
+   * associated with this Stream disconnects.
+   *
+   * @private
+   */
+  _onDisconnect() {
+    this.destroy();
+  }
+
+  /**
+   * Explicitly sets read operations to a no-op
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  _read() {
+  }
```

Comment:
You can remove the `eslint-disable` comment and add `return undefined` as the body of the function.